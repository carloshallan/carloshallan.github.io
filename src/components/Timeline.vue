<template>
  <div class="WorkTimeline">
    <div v-for="(job, index) in jobs" class="job" :key="index">
      <div class="header">
        <h2 class="title">{{ job.title }}</h2>
        <h3 class="subtitle">
          <span class="light-pink">{{ job.subtitle }}</span> {{ job.date }}
        </h3>
      </div>
      <ul class="jobDescription">
        <li>{{ job.jobDescription.description }}</li>
        <li>
          <strong class="description-title">{{
            job.jobDescription.title
          }}</strong>
          <ul>
            <li v-for="(stack, j) in job.jobDescription.listOfStack" :key="j">
              {{ stack.description }}
              <ul v-if="stack.children">
                <li v-for="(child, k) in stack.children" :key="k">
                  {{ child.description }}
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>
</template>
<script lang="ts">
import Vue, { PropType } from 'vue'
import { Job } from '@/types'

export default Vue.extend({
  name: 'workTimeline',
  props: {
    jobs: {
      type: Array as PropType<Array<Job>>
    }
  }
})
</script>
<style lang="stylus" scoped>
@import "../styles/default.styl"

.job
  display flex
  justify-content center
  align-items flex-start
  position relative
  gap 30px

  .header
    width 30%
    display flex
    flex-direction column
    align-items flex-end
    justify-content flex-end

  ul
    display flex
    flex-direction column
    font-size 20px

    li
      list-style none
      padding-bottom 20px

      ul
        margin-top 20px

      li
        list-style none
        padding-bottom 10px
        font-size 16px
        opacity 0.8

      li::before
        content '-> '
        FiraCode()
        color light-pink

.jobDescription
  width 40%
  display flex
  position relative
  padding 2px 40px

.jobDescription::after
.jobDescription::before
  content ''
  width 12px
  background-color light-pink
  display block
  position absolute
  border-radius 30px
  left 0px

.jobDescription::before
  top 12px
  height 12px !important

.jobDescription::after
  top 45px
  height calc(100% - 55px)

.title
  FiraSans()
  text-align right
  font-size 26px
  color green
</style>
