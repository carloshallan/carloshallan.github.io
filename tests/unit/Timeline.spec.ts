import { mount } from '@vue/test-utils'
import Timeline from '@/components/Timeline.vue'

describe('Timeline.vue', () => {
  it('Render timeline component', () => {
    const timeline = mount(Timeline)

    expect(timeline.element).toMatchSnapshot()
  })
})
