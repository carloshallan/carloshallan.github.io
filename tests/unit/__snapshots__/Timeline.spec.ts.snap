// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline.vue Render timeline component 1`] = `
<div
  class="WorkTimeline"
/>
`;
